.github/workflows/ci.yml

```
@@ -16,17 +21,67 @@
-        run: ./build/runTests
+        run: ./build/runTests --gtest_output="xml:test-results.xml"
+
+      - name: Generate code coverage report
+        run: |
+          cd build
+          lcov --capture --directory . --output-file coverage.info
+          lcov --remove coverage.info '/usr/*' --output-file coverage.info
+          lcov --list coverage.info
+
+      - name: Upload code coverage report to Codecov
+        uses: codecov/codecov-action@v3
```

actionlint (1.7.4)

the runner of "codecov/codecov-action@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)